File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void findPreOrder(TreeNode *node, vector<int> &ans) {
        if(node == NULL) 
            return;
        
        ans.push_back(node->val);

        findPreOrder(node->left, ans);

        findPreOrder(node->right, ans);


    }

    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> ans;
        findPreOrder(root, ans);
        return ans;
    }
};

File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        def traversal(node):
            if node is None:
                return
            
            ans.append(node.val)
            traversal(node.left)
            traversal(node.right)
            
        ans = [] 
        traversal(root)
        
        
        return ans